File: C#/DesignPatterns/AbstractFactory/AnimalsCreator.cs
using AbstractFactory.animals;

namespace AbstractFactory
{
    public class AnimalsCreator
	{
		public IEnumerable<IAnimal> CreateAnimals()
		{
			while (true)
			{
				var isCarnivore = Random.Shared.Next(0, 5) == 0;
				if (isCarnivore)
				{
					yield return new Wolf();
				}
				else
				{
					yield return new Bison();
				}
			}
		}
	}
}

File: Java/DesignPatterns/src/main/java/AbstractFactory/AnimalsCreator.java
package AbstractFactory;

import java.util.Iterator;
import java.util.Random;

public class AnimalsCreator {
    public Iterator<IAnimal> createAnimals() {
        return new Iterator<IAnimal>() {
            private Random random = new Random();

            @Override
            public boolean hasNext() {
                return true; // Infinite iterator
            }

            @Override
            public IAnimal next() {
                boolean isCarnivore = random.nextInt(5) == 0;
                if (isCarnivore) {
                    return new Wolf();
                } else {
                    return new Bison();
                }
            }
        };
    }
}
